stock_mts_mto_rule/models/stock_rule.py

```diff
@@ -36,14 +39,16 @@ def _check_mts_mto_rule(self):
                     ) % (rule.name,)
                     raise ValidationError(msg)
 
-    def get_mto_qty_to_order(self, product, product_qty, product_uom, values):
+    def get_mto_qty_to_order(
+        self, product, product_qty, product_uom, values, qty_unavailable=0.0
+    ):
         self.ensure_one()
         precision = self.env["decimal.precision"].precision_get(
             "Product Unit of Measure"
         )
         src_location_id = self.mts_rule_id.location_src_id.id
         product_location = product.with_context(location=src_location_id)
-        virtual_available = product_location.virtual_available
+        virtual_available = product_location.virtual_available - qty_unavailable
```

Comment:
I thought about this pull request again, and I noticed potential sources of error.

 First, we can't simply subtract the unavailable quantity from the virtual quantity. Since the source location can change with each rule, so we might subtract quantities from other locations.

And secondly, it is problematic to first calculate the quantity for each procurement and then execute it. Since, of course, each execution can lead to an increase in the virtual available quantity, I think that was @ferran-S73 initial problem.